lib/elixir/lib/macro.ex

```diff
@@ -315,6 +315,20 @@ defmodule Macro do
   def update_meta(other, _fun) do
     other
   end
+  @doc """
+  Generates AST nodes for a given number of required argument variables using
+  `Macro.var/2`.
+
+  ## Examples
+
+      iex> Macro.generate_arguments(2, __MODULE__)
+      [{:variable1, [], __MODULE__}, {:variable2, [], __MODULE__}]
+
+  """
+  def generate_arguments(number_of_args), do: generate_arguments(number_of_args, __MODULE__)
+  def generate_arguments(number_of_args, module) do 
```

Comment:
We should add guards here IMO. We can have `when is_integer(number_of_args) and number_of_args >= 0 and is_atom(module)` since `module` should be a module or `nil`, correct me if I am wrong. :)